### PR TITLE
Redux IV - redux-recompose

### PR DESCRIPTION
**Pull request by renzowolox**
## Summary
* redux-recompose added

## Trello Card
* https://trello.com/c/N6N0t7aM

### CODE REVIEW

File: react/tic-tac-toe/src/react-admin.d.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+declare module 'redux-recompose';
```
Comment (fededirocco34) [inline react/tic-tac-toe/src/react-admin.d.ts]: Add empty line

File: react/tic-tac-toe/src/redux/login/actions.js [hunk 2]
```diff
@@ -1,38 +1,50 @@
...
-    type: 'GET_USER_LOGGED'
+  setToken: token => ({
+    type: actions.SET_TOKEN,
+    target: 'token',
+    payload: token
   }),
-  setTokenFromLocalStorage: token => dispatch => {
-    dispatch(actionCreators.saveToken(token));
+  setSession: token => dispatch => {
+    dispatch(actionCreators.setToken(token));
```
Comment (fededirocco34) [inline react/tic-tac-toe/src/redux/login/actions.js]: For me its better to always do like this :point_down:. Try it if it works

```
setSession: token => ({
    type: actions.SET_TOKEN,
    target: 'token',
    payload: token,
    injections: [
      withSuccess( dispatch => dispatch(push(ROUTES.game)))
    ]
  })
```

File: react/tic-tac-toe/src/redux/login/actions.js [hunk 3]
```diff
@@ -1,38 +1,50 @@
...
     dispatch(push(ROUTES.game));
   },
   logout: () => dispatch => {
     localStorage.removeItem(TOKEN_GAME);
-    dispatch(actionCreators.clearToken());
+    dispatch(actionCreators.clearToken);
```
Comment (fededirocco34) [inline react/tic-tac-toe/src/redux/login/actions.js]: Same here, every action needs to have a `type`, `target` and `payload` so its easy  to read and understand. Try to recomponse this `logout` :laughing:

File: react/tic-tac-toe/src/redux/login/reducer.js [hunk 4]
```diff
@@ -1,30 +1,27 @@
...
+    [actions.SET_TOKEN]: (state, action) => ({
+      ...state,
+      token: action.payload,
+      isLogged: true
+    }),
+    [actions.CLEAR_TOKEN]: (state, action) => ({
+      ...state,
+      token: undefined
+    })
+  }
```
Comment (fededirocco34) [inline react/tic-tac-toe/src/redux/login/reducer.js]: For me its ok but do you read about `onReadValue()` and `onSpreadValue()`? Maybe you can use them

File: react/tic-tac-toe/src/redux/login/actions.js [hunk 5]
```diff
@@ -1,39 +1,53 @@
...
+  setSession: token => dispatch => {
+    dispatch(actionCreators.setToken(token));
     dispatch(push(ROUTES.game));
   },
-  logout: () => dispatch => {
+  logout: () => {
     localStorage.removeItem(TOKEN_GAME);
-    dispatch(actionCreators.clearToken());
-    dispatch(push(ROUTES.login));
+    /*dispatch(push(ROUTES.login));*/
```
Comment (erikankodelia) [inline react/tic-tac-toe/src/redux/login/actions.js]: Delete this comment

**Review by ghost — APPROVED** [review]

**Pull request merged**